### PR TITLE
Dvl/quentin/remote admin/search in message

### PR DESCRIPTION
**Pull request by quentinql**

### CODE REVIEW

**Review by Zluurk — APPROVED** [review]

File: pypeman/msgstore.py [hunk 1]
```diff
@@ -74,7 +74,26 @@ async def get_msg_content(self, id):
         :return: A dict `{'id':<message_id>, 'message_content': Message}`.
         """
 
-    async def search(self, start=0, count=10, order_by='timestamp', start_dt=None, end_dt=None):
+    async def is_regex_in_msg(self, id, rtext):
+        """
+        Return True if the str(msg) contains the regex rtext
+
+        :param id: Message id. Message store dependant.
+        :param rtext: String. The rtext is the text to convert in regex and to search in msg
```
Comment (feenes) [inline pypeman/msgstore.py:82]: rtext: string of regular expression  to search in msg

File: pypeman/msgstore.py [hunk 2]
```diff
@@ -74,7 +74,26 @@ async def get_msg_content(self, id):
...
+        :return: True if it matchs False otherwise
```
Comment (feenes) [inline pypeman/msgstore.py:83]: if it matches

File: pypeman/remoteadmin.py [hunk 3]
```diff
@@ -473,6 +491,12 @@ def do_list(self, channel, arg):
                 if arg == "--preview":
                     to_preview = True
                     args.remove(arg)
+                if arg.startswith("text="):
+                    text = arg.split("=")[1]
```
Comment (feenes) [inline pypeman/remoteadmin.py:495]: `arg.split("=", 1)[1]` to allow having `=` in search text

**Review by feenes — APPROVED** [review]

File: pypeman/msgstore.py (hunk 1 above)
Comment (quentinql) [reply pypeman/msgstore.py]: done

File: pypeman/msgstore.py (hunk 2 above)
Comment (quentinql) [reply pypeman/msgstore.py]: done

File: pypeman/msgstore.py [hunk 4]
```diff
@@ -74,7 +74,26 @@ async def get_msg_content(self, id):
...
+        """
+
+    async def is_txt_in_msg(self, id, text):
+        """
+        Return True if the str(msg) contains param text
+
+        :param id: Message id. Message store dependant.
+        :param text: String. The text to search in msg
+        :return: True if it text is find False otherwise
```
[the remark this answers is not in the document]
Comment (quentinql) [reply pypeman/msgstore.py]: done

File: pypeman/msgstore.py [hunk 5]
```diff
@@ -277,7 +405,25 @@ async def count_msgs(self):
                             count += 1
         return count
 
-    async def search(self, start=0, count=10, order_by='timestamp'):
+    async def is_regex_in_msg(self, id, rtext):
+        msg = await self.get_msg_content(id)
+        try:
+            msg.payload = str(msg.payload)[:1000]
```
[the remark this answers is not in the document]
Comment (quentinql) [reply pypeman/msgstore.py]: done

File: pypeman/remoteadmin.py (hunk 3 above)
Comment (quentinql) [reply pypeman/remoteadmin.py]: done

**Pull request merged**